client/src/app/pages/assessment/import-questionnaire-form/import-questionnaire-form.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from "react";
+import axios, { AxiosResponse } from "axios";
+import { useTranslation } from "react-i18next";
+import * as yup from "yup";
+
+import {
+  ActionGroup,
+  Button,
+  ButtonVariant,
+  Checkbox,
+  FileUpload,
+  Form,
+  FormGroup,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+} from "@patternfly/react-core";
+
+import { UPLOAD_FILE } from "@app/api/rest";
+import { getAxiosErrorMessage } from "@app/utils/utils";
+import { NotificationsContext } from "@app/components/NotificationsContext";
+import { HookFormPFGroupController } from "@app/components/HookFormPFFields";
+import { useForm } from "react-hook-form";
+import { FileLoadError, IReadFile } from "@app/api/models";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useCreateFileMutation } from "@app/queries/targets";
+
+export interface ImportQuestionnaireFormProps {
+  onSaved: (response: AxiosResponse) => void;
+}
+export interface ImportQuestionnaireFormValues {
+  yamlFile: IReadFile;
+}
+
+export const yamlFileSchema: yup.SchemaOf<IReadFile> = yup.object({
+  fileName: yup.string().required(),
+  fullFile: yup.mixed<File>(),
+  loadError: yup.mixed<FileLoadError>(),
+  loadPercentage: yup.number(),
+  loadResult: yup.mixed<"danger" | "success" | undefined>(),
+  data: yup.string(),
+  responseID: yup.number(),
+});
+
+export const ImportQuestionnaireForm: React.FC<
+  ImportQuestionnaireFormProps
+> = ({ onSaved }) => {
+  const { t } = useTranslation();
+  const { pushNotification } = React.useContext(NotificationsContext);
+
+  const [filename, setFilename] = React.useState<string>();
+  const [isFileRejected, setIsFileRejected] = useState(false);
+  const validationSchema: yup.SchemaOf<ImportQuestionnaireFormValues> = yup
+    .object()
+    .shape({
+      yamlFile: yamlFileSchema,
+    });
+  const methods = useForm<ImportQuestionnaireFormValues>({
+    resolver: yupResolver(validationSchema),
+    mode: "onChange",
+  });
+
+  const {
+    handleSubmit,
+    formState: { isSubmitting, isValidating, isValid, isDirty },
+    getValues,
+    setValue,
+    control,
+    watch,
+    setFocus,
+    clearErrors,
+    trigger,
+    reset,
+  } = methods;
+
+  const { mutateAsync: createYamlFileAsync } = useCreateFileMutation();
+
+  const handleFileUpload = async (file: File) => {
+    setFilename(file.name);
+    const formFile = new FormData();
+    formFile.append("file", file);
+
+    const newYamlFile: IReadFile = {
+      fileName: file.name,
+      fullFile: file,
+    };
+
+    return createYamlFileAsync({
+      formData: formFile,
+      file: newYamlFile,
+    });
+  };
+
+  const onSubmit = (values: ImportQuestionnaireFormValues) => {
+    console.log("values", values);
+  };
+  return (
+    <Form onSubmit={handleSubmit(onSubmit)}>
+      <HookFormPFGroupController
+        control={control}
+        name="yamlFile"
```

Comment:
I see many "yamlFile" occurences here, maybe extract it to a const?

Reply:
I have moved to using the "name" provided by the controller component on subsequent occurrences. TS helps us here by inferring the types from the react hook form form values initialization. 


![Screenshot 2023-08-23 at 11 20 21 AM](https://github.com/konveyor/tackle2-ui/assets/11218376/60f921c0-eba1-4ec0-9f6a-c90bdc3a2a7c)

Reply:
I meant maybe to add on top of file something like: `const yamlID = "yamlFile"` or similar const name so if the type changes in the future, you can change it in only one place and not multiple places. If it's not possible because of the type inferring than please disregard this.

Reply:
I meant maybe to add on top of file something like: `const yamlID = "yamlFile"` or similar const name so if the type changes in the future, you can change it in only one place and not multiple places. If it's not possible because of the type inferring than please disregard this.

Reply:
The form values are defined in one place so I think that accomplishes what you are suggesting here. When we define the formValues, the internal TS support for react-hook-form allows us to infer the types from usage so that we will get a TS error if there is a spelling error or another issue. 

<img width="618" alt="Screenshot 2023-08-24 at 9 13 41 AM" src="https://github.com/konveyor/tackle2-ui/assets/11218376/52af0eb5-f229-4076-b640-4f46584a48a7">
<img width="721" alt="Screenshot 2023-08-24 at 9 13 33 AM" src="https://github.com/konveyor/tackle2-ui/assets/11218376/449e0460-4f46-4ddc-99e0-a854a510d7a2">

Reply:
gotcha, thanks for the explanation :smile: